datastore.go

```diff
@@ -275,17 +311,34 @@ func (d *Datastore) Batch() (ds.Batch, error) {
 	return tx, nil
 }
 
-func (d *Datastore) CollectGarbage() error {
+func (d *Datastore) CollectGarbage() (err error) {
 	d.closeLk.RLock()
 	defer d.closeLk.RUnlock()
 	if d.closed {
 		return ErrClosed
 	}
 
-	err := d.DB.RunValueLogGC(d.gcDiscardRatio)
+	gcTimeout := time.NewTimer(d.maxGcDuration)
+	defer gcTimeout.Stop()
+
+LOOP:
+	for {
+		select {
+		case <-gcTimeout.C:
+			break LOOP
+		default:
+			if err == nil {
+				err = d.DB.RunValueLogGC(d.gcDiscardRatio)
```

Comment:
Do I understand this correctly as by default running RunValueLogGC() repeteadly for 1 minute non-stop? I'm not sure this works. In order to timeout a GC operation this would need support from badger.